まだ家計簿アプリビルドエラーなんだけど

@Claude
まだ家計簿アプリビルドエラーなんだけど！なおして

Failed to compile.

./app/household/page.tsx:345:25
Type error: Type '(entry: { category: string; amount: number; }) => string' is not assignable to type 'PieLabel | undefined'.
  Type '(entry: { category: string; amount: number; }) => string' is not assignable to type '(props: PieLabelRenderProps) => ReactNode | ReactElement<SVGElement, string | JSXElementConstructor<any>>'.
    Types of parameters 'entry' and 'props' are incompatible.
      Type 'PieLabelRenderProps' is missing the following properties from type '{ category: string; amount: number; }': category, amount

  343 |                         cy="50%"
  344 |                         outerRadius={100}
> 345 |                         label={(entry: { category: string; amount: number }) => `${entry.category}: ${formatCurrency(entry.amount)}`}
      |                         ^
  346 |                         labelLine={false}
  347 |                       >
  348 |                         {expenseByCategory.map((entry, index) => (
Next.js build worker exited with code: 1 and signal: null
Error: Process completed with exit code 1.

--- a/app/household/page.tsx
+++ b/app/household/page.tsx
@@ -342,7 +342,10 @@ export default function HouseholdPage() {
                         cx="50%"
                         cy="50%"
                         outerRadius={100}
-                        label={(entry: { category: string; amount: number }) => `${entry.category}: ${formatCurrency(entry.amount)}`}
+                        label={(props) => {
+                          const entry = expenseByCategory[props.index];
+                          return `${entry.category}: ${formatCurrency(entry.amount)}`;
+                        }}
                         labelLine={false}
                       >
                         {expenseByCategory.map((entry, index) => (
@@ -370,7 +373,10 @@ export default function HouseholdPage() {
                         cx="50%"
                         cy="50%"
                         outerRadius={100}
-                        label={(entry: { category: string; amount: number }) => `${entry.category}: ${formatCurrency(entry.amount)}`}
+                        label={(props) => {
+                          const entry = incomeByCategory[props.index];
+                          return `${entry.category}: ${formatCurrency(entry.amount)}`;
+                        }}
                         labelLine={false}
                       >
                         {incomeByCategory.map((entry, index) => (

